File: app/src/main/java/ru/magflayer/spectrum/presentation/common/model/PageAppearance.java
package ru.magflayer.spectrum.presentation.common.model;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PageAppearance {

    private Boolean showFloatingButton;

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        private PageAppearance pageAppearance = new PageAppearance();


        public Builder showFloatingButton(final Boolean showFloatingButton) {
            pageAppearance.setShowFloatingButton(showFloatingButton);
            return this;
        }

        public PageAppearance build() {
            return pageAppearance;
        }

    }
}

File: app/src/main/java/ru/magflayer/spectrum/presentation/common/model/PageAppearance.kt
package ru.magflayer.spectrum.presentation.common.model

data class PageAppearance(
        var floatingButtonState: FloatingButtonState = FloatingButtonState.INVISIBLE
) {

    class Builder {
        private val pageAppearance = PageAppearance()


        fun showFloatingButton(floatingButtonState: FloatingButtonState): Builder {
            pageAppearance.floatingButtonState = floatingButtonState
            return this
        }

        fun build(): PageAppearance {
            return pageAppearance
        }

    }

    companion object {
        fun builder(): Builder {
            return Builder()
        }
    }

    enum class FloatingButtonState {
        VISIBLE, INVISIBLE, NO_INFLUENCE
    }
}
